### PR TITLE
Test methods with reserved regex chars fail to rerun

### LINKED ISSUE #15: Test methods with reserved regex chars fail to rerun
A dynamic test method containing  reserved regex chars in a parametrized data spock test failes to execute. 

```
    @spock.lang.Unroll
    def "unrolled with param [#param]"() {
        expect:
        result

        where:
        param << ['foo', 'bar', 'baz']
        result << [true, false, true]
    }
```

Outputs:
```
* What went wrong:
Execution failed for task ':test'.
> org.gradle.test-retry was unable to retry the following test methods, which is unexpected. Please file a bug report at https://github.com/gradle/test-retry-gradle-plugin/issues
     org.gradle.testretry.sample.SampleTests#unrolled with param [bar]
```

Buildscan: https://gradle.com/s/3lqzuva3356sc

### PR DESCRIPTION
**Pull request by jonnybbb**
resolves #15

### CODE REVIEW

File: plugin/src/test/groovy/org/gradle/testretry/SpockFuncTest.groovy [hunk 1]
```diff
@@ -123,6 +123,16 @@ class SpockFuncTest extends AbstractTestFrameworkPluginFuncTest {
                     param << ['foo', 'bar', 'baz']
                     result << [true, false, true]
                 }
+
+                @spock.lang.Unroll
+                def "unrolled with param [#param]"() {
```
Comment (guylabs) [inline plugin/src/test/groovy/org/gradle/testretry/SpockFuncTest.groovy]: Please add a separate test for the escaping and create a method that includes all Regex chars such that we are sure that we properly escape all of these chars.

File: plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java [hunk 2]
```diff
@@ -56,6 +68,22 @@ public void visitEnd() {
     public String getTestMethodName() {
         return testMethodName;
     }
+
+    private static String escapeRegEx(String aRegexFragment) {
```
[the remark this answers is not in the document]
Comment (jonnybbb) [reply plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java]: yes I did - but it escapes the whole pattern with`\Q .... \E`. Grouping via regex for several params seems to be way too complex.

[the remark this answers is not in the document]
Comment (guylabs) [reply plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java]: But the `methodPattern` lambda argument is not in fact a regex, its the name of the param method as stated here: https://github.com/gradle/test-retry-gradle-plugin/blob/5bd345de6b825df29e61b566651cffffceceb2af/plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java#L114 Or do I miss something here @breskeby ?

File: plugin/src/test/groovy/org/gradle/testretry/SpockFuncTest.groovy (hunk 1 above)
Comment (jonnybbb) [reply plugin/src/test/groovy/org/gradle/testretry/SpockFuncTest.groovy]: added a dedicated regex test

File: plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (guylabs) [reply plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java]: All good as discussed with @jonnybbb

[the remark this answers is not in the document]
Comment (jonnybbb) [reply plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java]: as discussed with @guylabs - can be resolved

**Comment by jkschneider** [comment]
Thanks @jonnybbb! Good catch.

**Pull request merged**